usb_test.go

```diff
@@ -94,3 +96,35 @@ func TestOpenDeviceWithVIDPID(t *testing.T) {
 		}
 	}
 }
+
+func TestOpenDeviceWithFileDescriptor(t *testing.T) {
+	ctx := newContextWithImpl(newFakeLibusb())
+	defer ctx.Close()
+
+	// file descriptor is an index to the FakeDevices array
+	for ix, d := range []struct {
+		vid, pid ID
+		exists   bool
+	}{
+		{0x9999, 0x0001, true},
+		{0x8888, 0x0002, true},
+		{0x1111, 0x1111, true},
+		{0x7777, 0x0003, false},
+	} {
+		// The fake keeps the order of fakeDevices so that it can be indexed by a fake "fileDescriptor"
+		dev, err := ctx.OpenDeviceWithFileDescriptor(uintptr(ix))
```

Comment:
this conflates the ordering of the test cases with the file descriptor values, I think it's fragile. See my other comment about a map for sys devices.